midpoint/install/system-requirements/index.adoc

```diff
@@ -223,93 +230,123 @@ Refer to the xref:/midpoint/release/[midPoint Releases] documentation for softwa
-When a node is down, other nodes need to be able to replace all of its jobs.
+==== Resources
 
-End users and identity engineers use midPoint Web UI over a load balancer (HTTPS).
+Then there are the resources.
+There are many different communication protocols the resources may use.
+The schema above shows just a couple of the most common ones.
 
-You can check that connections to all source or target systems are working using the following tools: `ping`, `telnet`, and `wget`.
-You can use them from the midPoint server and also from the workstation (once tunnels have been configured).
+You may have an HR system which is only able to give midPoint CSV files, there may be a more complex system, data of which you can access using SQL, you may need to connect an Active Directory, and so on.
+With some resources, such as xref:/connectors/connectors/com.evolveum.polygon.connector.sap.SapConnector/[SAP and JCo], you need to enable API on the target resource, open your firewall on servers where the target system is running, and create an account with appropriate permissions to manage identities.
+In certain cases, you may have to deploy a <<connector_servers_sizing,connector server>> to access some special resources.
+The list of options is endless and very much depends on what exactly you need to manage with midPoint.
+
+=== Multi-Node Deployment
+
+The situation gets a bit more complex when you deploy midPoint on two or more nodes:
+
+//image::environment-schema-HA.png[]
+image::midpoint-environment-schema-HA-two-nodes.svg[title="Deployment of midPoint with two nodes, load balancer, VPN, and a few example resources"]
+
+==== Work Distribution
+
+In multi-node deployment, the job distribution among the nodes is handled centrally.
+
+The repository DB keeps track of the task states, i.e., which are to be done, in progress, or done.
+This helps to ensure that no single task is processed by two nodes concurrently.
+The Quartz library then serves as a smart timer, creating worker tasks which it gives to nodes for processing.
+The available tasks are picked by the nodes on the first-come-first-served basis, which ensures reasonably even task distribution.
+
+Refer to xref:/midpoint/reference/tasks/task-manager/[] for details on task handling in midPoint.
+
+==== Communication Among Nodes
+
+Primarily, the nodes communicate with the central Quartz library rather than among themselves,
+with the exception of *cache invalidation*.
+When a node changes data in the midPoint database, the node informs other nodes about the need to invalidate their cache.
+The *communication between nodes runs over HTTPS*.
+
+==== Node Management Access
+
+Identity engineers need to access all nodes, either using the web UI or the REST API, be it via a load balancer or directly. 
+Depending on the setup, they may need to connect to the nodes through a VPN which tunnels them to a particular node or the load balancer.
```

Comment:
Connecting to a particular node might be useful only for debugging. Otherwise nodes are equal, so even engineers don't care which they are connected to.